### ISSUE TITLE
Dirty working directory not being correctly detected when tags are missing

### ISSUE THREAD

**Issue by kinghajj** [pypa/setuptools-scm#86]
I'm working with a "big repo" with many libraries/services. One service, "tasks," references the "model" library in its requirements.txt

```
../lib/model
```

If I run

```
$ pex -r requirements.txt -o test.pex
```

Then the generated wheel in ~/.pex/build is called `model-0.1.dev239+n6e5cad8-py2-none-any.whl`, which doesn't have the timestamp that the documentation suggests, despite the fact that the repository is dirty:

```
Changes to be committed:
  (use "git reset HEAD <file>..." to unstage)

    modified:   lib/model/setup.py
    modified:   workers/tasks/requirements.txt
```

**Comment by RonnyPfannschmidt** [comment]
at first glace that looks like a problem with git describe interpretation

whats the output of git describe? also please run the `SETUPTOOLS_SCM_DEBUG=1' python setup.py --version`

**Comment by kinghajj** [comment]
```
fatal: No names found, cannot describe anything.
```

Which apparently is caused by not having any tags--which is expected, since this is a fairly new repo. Maybe that requirement should be mentioned in the documentation? Thanks for the help.

**Comment by RonnyPfannschmidt** [comment]
no, thats actually a bug - the handler for nothign to describe doesnt check if there is changes in the workdir

**Comment by RonnyPfannschmidt** [comment]
closed by   d041864
